neurom/core/morphology.py

```diff
@@ -439,13 +442,16 @@ def _subtrees_type(self):
             if section.type != section.parent.type:
                 subtree_types.append(section.to_morphio().type)
 
-        if len(subtree_types) == 1:
-            return subtree_types[0]
+        return NeuriteType(subtree_types)
+        # return NeuriteType.from_list(subtree_types)
```

Comment:
So the issue with this is that we want to allow specific heterogeneous types, otherwise we will have heterogeneous types for neurites that have section type artifacts.

Reply:
Well, I think the user should be responsible of the quality of the morphologies he is analyzing. IMO we should just provide some functions to check and fix heterogeneous types, but then the analysis functions should just consider that the morphology is clean.
Nevertheless, we could issue warnings if the heterogeneous type looks weird (e.g. `[axon, soma, apical, basal]` :sweat_smile: ).

Reply:
The goal of adding heterogemenous types in NeuroM is not to expose impurities at the section level as neurite subtypes. MorphIO is much better equipped to handle that given that it is purely section-centered.

NeuroM should support valid mixed neurite types that have been found in the literature. I am not against allowing the user to specify a custom mixed type, but I believe it's not a good idea for neurom to return mixed subtypes due to impurities.

Reply:
We can restrict the supported combinations if you wish. And thus NeuroM would just raise an exception when an unknown subtype is found?

Reply:
yeah